File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test2/TestApplicationComponent.java
package it.droidcon.testingdaggerrxjava.test2;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import it.droidcon.testingdaggerrxjava.dagger.ApplicationComponent;
import it.droidcon.testingdaggerrxjava.dagger.StackOverflowServiceModule;
import it.droidcon.testingdaggerrxjava.userlist.UserListModuleDef;
import javax.inject.Singleton;

@Singleton
@Component(modules = {
    TestUserInteractorModule.class,
    StackOverflowServiceModule.class,
    AndroidInjectionModule.class,
    UserListModuleDef.class
})
public interface TestApplicationComponent
    extends ApplicationComponent {
  void inject(EndToEndTest endToEndTest);
}

File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test2/TestApplicationComponent.kt
package it.droidcon.testingdaggerrxjava.test2

import dagger.Component
import it.droidcon.testingdaggerrxjava.dagger.ApplicationComponent
import it.droidcon.testingdaggerrxjava.dagger.StackOverflowServiceModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(TestUserInteractorModule::class, StackOverflowServiceModule::class))
interface TestApplicationComponent : ApplicationComponent {
    fun inject(userListActivityTest: UserListActivityTest)
}
